### PR TITLE
Acceptance test for deleting/restoring files by sharer and sharee and checking activity

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

Add acceptance tests for:

- deleting a shared file/folder and check activity on both sides.
- deleting a shared file/folder, restore on each side, and check activity on both sides

## Related Issue

https://github.com/owncloud/activity/issues/747

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiActivity/list.feature [hunk 1]
```diff
@@ -433,3 +433,160 @@ Feature: List activity
...
+
+  @skipOnOcV10.2
+  Scenario: Sharer and sharee check activity after sharer deletes shared file
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | user0    |
+      | user1    |
+    And user "user0" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
+    And user "user0" has shared file "/lorem.txt" with user "user1"
+    When user "user0" deletes folder "/lorem.txt" using the WebDAV API
```
Comment (dpakach) [inline tests/acceptance/features/apiActivity/list.feature]:
```suggestion
    When user "user0" deletes file "/lorem.txt" using the WebDAV API
```
Still works but would be more readable this way

File: tests/acceptance/features/apiActivity/list.feature [hunk 2]
```diff
@@ -433,3 +433,160 @@ Feature: List activity
...
+
+  @skipOnOcV10.2
+  Scenario: Sharer and sharee check activity after sharee deletes shared file
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | user0    |
+      | user1    |
+    And user "user0" has uploaded file "filesForUpload/lorem.txt" to "/lorem.txt"
+    And user "user0" has shared file "/lorem.txt" with user "user1"
+    When user "user1" deletes folder "/lorem.txt" using the WebDAV API
```
Comment (dpakach) [inline tests/acceptance/features/apiActivity/list.feature]: Same as above

File: tests/acceptance/features/webUIActivityDeleteRestore/delete.feature [hunk 3]
```diff
@@ -146,3 +146,34 @@ Feature: Deleted files/folders activities
...
+  Scenario: Sharer and sharee check activity after sharer deletes shared file
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | user1    |
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    When the user reloads the current page of the webUI
+    And user "user0" has deleted file "textfile0.txt"
+    When the user browses to the activity page
+    Then the activity number 1 should contain message "You deleted textfile0.txt" in the activity page
+    And the activity number 2 should contain message "You shared textfile0.txt with UUser One" in the activity page
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUIActivityDeleteRestore/delete.feature]:
```suggestion
    And the activity number 2 should have a message saying that you have shared folder "textfile0.txt" with user "User One"
```

File: tests/acceptance/features/webUIActivityDeleteRestore/delete.feature [hunk 4]
```diff
@@ -146,3 +146,34 @@ Feature: Deleted files/folders activities
...
+    When the user re-logs in as "user1" using the webUI
+    When the user browses to the activity page
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUIActivityDeleteRestore/delete.feature]:
```suggestion
    And the user browses to the activity page
```

File: tests/acceptance/features/webUIActivityDeleteRestore/delete.feature [hunk 5]
```diff
@@ -146,3 +146,34 @@ Feature: Deleted files/folders activities
...
+      | username |
+      | user1    |
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    And user "user1" has deleted file "textfile0.txt"
+    And the user re-logs in as "user1" using the webUI
+    When the user browses to the activity page
+    Then the activity number 1 should contain message "You unshared textfile0.txt shared by UUser Zero from self" in the activity page
+    And the activity number 2 should contain message "User Zero shared textfile0.txt with you" in the activity page
+    When the user re-logs in as "user0" using the webUI
+    When the user browses to the activity page
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUIActivityDeleteRestore/delete.feature]:
```suggestion
    And the user browses to the activity page
```

File: tests/acceptance/features/webUIActivityDeleteRestore/restore.feature [hunk 6]
```diff
@@ -247,3 +247,36 @@ Feature: Restored files/folders activities
...
+      | username |
+      | user1    |
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    When the user reloads the current page of the webUI
+    And user "user0" has deleted file "textfile0.txt"
+    And user "user0" has restored the file with original path "textfile0.txt"
+    When the user browses to the activity page
+    Then the activity number 1 should contain message "You restored textfile0.txt" in the activity page
+    And the activity number 2 should contain message "You deleted textfile0.txt" in the activity page
+    And the activity number 3 should contain message "You shared textfile0.txt with UUser One" in the activity page
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUIActivityDeleteRestore/restore.feature]:
```suggestion
    And the activity number 3 should have a message saying that you have shared file "textfile0.txt" with user "User One"
```

File: tests/acceptance/features/webUIActivityDeleteRestore/restore.feature [hunk 7]
```diff
@@ -247,3 +247,36 @@ Feature: Restored files/folders activities
...
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    And the user re-logs in as "user1" using the webUI
+    And user "user1" has deleted file "textfile0.txt"
+    And user "user0" has restored the file with original path "textfile0.txt"
+    When the user browses to the activity page
+    Then the activity number 1 should contain message "You unshared textfile0.txt shared by UUser Zero from self" in the activity page
+    And the activity number 2 should contain message "User Zero shared textfile0.txt with you" in the activity page
+    When the user re-logs in as "user0" using the webUI
+    When the user browses to the activity page
+    Then the activity number 1 should contain message "You shared textfile0.txt with UUser One" in the activity page
```
Comment (paurakhsharma) [inline tests/acceptance/features/webUIActivityDeleteRestore/restore.feature]:
```suggestion
    Then the activity number 1 should have a message saying that you have shared folder "textfile0.txt" with user "User One"
```

File: tests/acceptance/features/bootstrap/WebUIActivityContext.php [hunk 8]
```diff
@@ -261,6 +266,28 @@ public function theActivityNumberShouldHaveAMessageSayingSharedFolderWithUser(
 		PHPUnit\Framework\Assert::assertEquals($message, $latestActivityMessage);
 	}
 
+	/**
+	 * @When the activity number :index should have a message saying that you have unshared file/folder :entry shared by :user from self
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/WebUIActivityContext.php]: s/When/Then

File: tests/acceptance/features/webUIActivityDeleteRestore/delete.feature [hunk 9]
```diff
@@ -146,3 +146,34 @@ Feature: Deleted files/folders activities
...
+
+  @skipOnOcV10.2
+  Scenario: Sharer and sharee check activity after sharee deletes shared file
+    Given these users have been created with default attributes and without skeleton files:
+      | username |
+      | user1    |
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    And the user re-logs in as "user1" using the webUI
+    When the user deletes file "textfile0.txt" using the webUI
+    When the user browses to the activity page
```
Comment (phil-davis) [inline tests/acceptance/features/webUIActivityDeleteRestore/delete.feature]: s/When/And

File: tests/acceptance/features/webUIActivityDeleteRestore/delete.feature [hunk 10]
```diff
@@ -146,3 +146,34 @@ Feature: Deleted files/folders activities
...
+    Then the activity number 1 should have a message saying that you have unshared file "textfile0.txt" shared by "User Zero" from self
+    And the activity number 2 should contain message "User Zero shared textfile0.txt with you" in the activity page
+    When the user re-logs in as "user0" using the webUI
+    And the user browses to the activity page
+    And the activity number 1 should have a message saying that you have shared file "textfile0.txt" with user "User One"
```
Comment (phil-davis) [inline tests/acceptance/features/webUIActivityDeleteRestore/delete.feature]: s/And/Then

File: tests/acceptance/features/webUIActivityDeleteRestore/restore.feature [hunk 11]
```diff
@@ -247,3 +247,36 @@ Feature: Restored files/folders activities
...
+      | user1    |
+    And user "user0" has shared file "textfile0.txt" with user "user1"
+    And the user re-logs in as "user1" using the webUI
+    And user "user1" has deleted file "textfile0.txt"
+    And user "user0" has restored the file with original path "textfile0.txt"
+    When the user browses to the activity page
+    Then the activity number 1 should have a message saying that you have unshared file "textfile0.txt" shared by "User Zero" from self
+    And the activity number 2 should contain message "User Zero shared textfile0.txt with you" in the activity page
+    When the user re-logs in as "user0" using the webUI
+    When the user browses to the activity page
```
Comment (phil-davis) [inline tests/acceptance/features/webUIActivityDeleteRestore/restore.feature]: s/When/And

File: tests/acceptance/features/webUIActivityDeleteRestore/restore.feature [hunk 12]
```diff
@@ -247,3 +247,36 @@ Feature: Restored files/folders activities
...
+    And the activity number 1 should have a message saying that you have shared file "textfile0.txt" with user "User One"
```
Comment (phil-davis) [inline tests/acceptance/features/webUIActivityDeleteRestore/restore.feature]: s/And/Then

**Pull request merged**